dbt/adapters/athena/impl.py

```diff
@@ -57,6 +57,48 @@ def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
     def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
+    # TODO: Add more lf-tag unit tests when moto supports lakeformation
+    # moto issue: https://github.com/getmoto/moto/issues/5964
+    @available
+    def add_lf_tags(self, database: str, table: str = None, lf_tags: Dict[str, str] = None):
+        conn = self.connections.get_thread_connection()
+        client = conn.handle
+
+        lf_tags = lf_tags or conn.credentials.lf_tags
+        if not lf_tags:
+            logger.debug("No LF tags configured")
+            return
+
+        resource = {
+            "Database": {"Name": database},
+        }
+
+        if table:
+            resource = {
+                "Table": {
+                    "DatabaseName": database,
+                    "Name": table,
+                }
+            }
+
+        with boto3_client_lock:
+            lf_client = client.session.client(
+                "lakeformation", region_name=client.region_name, config=get_boto3_config()
+            )
+
+        lf_client.add_lf_tags_to_resource(
```

Comment:
I found an issue here. Save the result in a response and check if `Failures` key is available, in case is there, raise an exception with the failure error message. without that we have silent failures...

Reply:
Aah yes! Good catch, adding logic to deal with failures in my next commit

Reply:
@nicor88 I ended up logging each individual failure before throwing a general error, let me know if you had something else in mind